src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,87 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_SIZE = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int MULTIPLIER = 2;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size > table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        Node<K, V> newNode = new Node<>(key, value, null);
+        while (currentNode != null) {
+            if (currentNode.key == key
+                    || (currentNode.key != null && currentNode.key.equals(key))) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        table[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (key == node.key || (node.key != null && node.key.equals(key))) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] oldtable = table;
+        table = new Node[table.length * MULTIPLIER];
```

Comment:
When resizing the table, use the diamond operator with the correct generic types when initializing the new array. Replace with 'new Node<>(oldtable.length * MULTIPLIER);' to avoid unchecked assignment warnings.